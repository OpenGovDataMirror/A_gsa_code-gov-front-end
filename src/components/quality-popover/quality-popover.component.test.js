import React from 'react'
import { shallow } from 'enzyme'
import QualityPopover from 'components/quality-popover/quality-popover.component'

const props = {}

const popoverToggle = jest.fn()

let wrapper
let instance
describe('components - QualityPopover', () => {
  beforeEach(() => {
    wrapper = shallow(<QualityPopover {...props} />)
    instance = wrapper.instance()
  })

  describe('onClick', () => {
    it('should toggle `activated`', () => {
      document.body.innerHTML =
        '<div id="data-quality-popover" className="padding-top-3" hidden> </div>'
      instance.onClick()
      expect(wrapper.state('activated')).toBeTruthy()
      instance.onClick()
      expect(wrapper.state('activated')).toBeFalsy()
    })

    it('should toggle popover', () => {
      document.body.innerHTML =
        '<div id="data-quality-popover" className="padding-top-3" hidden> </div>'
      instance.onPopoverToggle()
      expect(
        wrapper.findWhere(x => x.prop('hidden') && x.prop('id').match('data-quality-popover'))
          .length
      ).toBe(1)
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
