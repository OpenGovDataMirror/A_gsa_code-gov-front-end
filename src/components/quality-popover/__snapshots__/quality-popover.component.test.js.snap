// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - QualityPopover render should render correctly 1`] = `
<div
  className="tablet-lg:padding-top-0 padding-top-2"
>
  <div
    className="pin-right tablet-lg:padding-right-1"
  >
    <span
      className="data-quality-title text-bold font-body-2xs"
      id="data-quality-title"
    >
      Data Quality Score
    </span>
    <div
      aria-labelledby="data-quality-title"
      className="icon usa-button bg-white icon-help-circled popper display-inline text-primary padding-x-05 padding-y-0 margin-left-2px"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    />
  </div>
  <div
    className="padding-top-3"
    hidden={true}
    id="data-quality-popover"
  >
    <div
      className="z-top bg-base-lightest padding-x-2 padding-y-105 radius-lg margin-bottom-neg-2 font-body-2xs line-height-sans-3"
    >
      The Data Quality Score is determined by using the information provided by Agencies in their 
      <Connect(CustomLinkComponent)
        to="/about/compliance/inventory-code"
      >
        code.json
      </Connect(CustomLinkComponent)>
       and by factors such as completeness and adherence to the
       
      <Connect(CustomLinkComponent)
        to="/federal-agencies/compliance/inventory-code"
      >
        metadata schema
      </Connect(CustomLinkComponent)>
      .
    </div>
  </div>
</div>
`;
